src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java

```diff
@@ -293,28 +288,22 @@ public ThreadRecord getThreadRecord(DcLot summary, DcChat chat) { // adapted fro
    **********************************************************************************************/
 
   private final Object threadsCritical = new Object();
+  private final Object incLoopsCritical= new Object();
 
-  private boolean imapThreadStartedVal;
-  private final Object imapThreadStartedCond = new Object();
   public Thread imapThread = null;
   private PowerManager.WakeLock imapWakeLock = null;
+  private int inboxLoops = 0;
```

Comment:
you could also use some AtomicInt's here to get rid of some synchronized blocks (except the one at line 402)

Reply:
didn't know that even exists :)
but i do no think this will block things?

Reply:
yes, AtomicIntegers have blocking read/write access: https://developer.android.com/reference/java/util/concurrent/atomic/AtomicInteger and can be used for programming concurrenency.